File: 513-find-bottom-left-tree-value/513-find-bottom-left-tree-value.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        return solve(root);
    }
private:
    int solve(TreeNode* root) {
        queue<TreeNode*> queue;
        queue.push(root);
        int ans = root -> val;
        
        while(!queue.empty()) {
            int levelSize = queue.size();
            
            for(int i = 0; i < levelSize; ++i) {
                auto currNode = queue.front();
                queue.pop();
                
                if(i == 0)
                    ans = currNode -> val;
                if(currNode -> left != nullptr) queue.push(currNode -> left);
                if(currNode -> right != nullptr) queue.push(currNode -> right);
            }
        }
        return ans;
    }
};

File: 513-find-bottom-left-tree-value/513-find-bottom-left-tree-value.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        queue = deque()
        queue.append(root)
        ans = root.val
        
        while queue:
            levelSize = len(queue)
            
            for i in range(levelSize):
                currNode = queue.popleft()
                
                if i == 0:
                    ans = currNode.val
                
                if currNode.left != None: queue.append(currNode.left)
                if currNode.right != None: queue.append(currNode.right)
        return ans